topo/topo.go

```diff
@@ -594,16 +594,20 @@ func (m *Manager) push(ctx context.Context) error {
 }
 
 func updateServicePortName(s *tpb.Service, port uint32) {
-	if s.Name != "" {
-		return
-	}
+	i := 0
 	for _, name := range s.Names {
 		if name != "" {
-			s.Name = name
-			return
+			s.Names[i] = name
+			i++
 		}
 	}
```

Comment:
I don't think this will work quite right.  The last element(s) can be repeated.  It is best to just create a new slice and then assign s.Names to that slice.

Reply:
Clarified the working of the snippet.